### PR TITLE
[CP-938] The e2e test for update functionality

### PR DESCRIPTION
**Pull request by igorbezsmertnyi**
Jira: [CP-938]

**Description**
This PR contains the implementation of the E2E testing for MuditaOS update functionality on Mudita Center
To be able to run the test, please provide the next environment variables:
TEST_INITIAL_VERSION=1.1.5 (make sure that your phone have 1.1.5v)
TEST_BINARY_PATH=../../some directory/mac/Mudita Center.app/Contents/MacOS/Mudita Center
TEST_GITHUB_USER=
TEST_GITHUB_PASSWORD

Connect the phone to the computer and run `npm run test:e2e`

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-936]: https://appnroll.atlassian.net/browse/CP-936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CP-938]: https://appnroll.atlassian.net/browse/CP-938?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by wercix** [comment]
Please, check task number one more time 😄

File: packages/e2e/helpers/github.helper.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class GitHubHelper {
+  private gitHubInstance: AxiosInstance
+
+  constructor() {
+    this.gitHubInstance = axios.create({
+      baseURL: "https://api.github.com/repos/mudita/MuditaOS",
+      auth: {
+        username: process.env.TEST_GITHUB_USER,
+        password: process.env.TEST_GITHUB_PASSWORD,
+      },
```
Comment (dkarski) [inline packages/e2e/helpers/github.helper.ts:12]: 🟡
I suggest using token

Tokens offer [a number of security benefits](https://github.blog/2020-07-30-token-authentication-requirements-for-api-and-git-operations/#background) over password-based authentication:

Unique – tokens are specific to GitHub and can be generated per use or per device
Revocable – tokens can can be individually revoked at any time without needing to update unaffected credentials
Limited – tokens can be narrowly scoped to allow only the access necessary for the use case
Random – tokens are not subject to the types of dictionary or brute force attempts that simpler passwords that you need to remember or enter regularly might be

File: packages/e2e/specs/overview/check-for-update.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import GitHubHelper from "../../helpers/github.helper"
+
+describe("Overview Page", () => {
+  describe("Check for update modal", () => {
+    before(async () => {
+      // Waiting for device connected through USB
+      await browser.executeAsync((done) => {
+        setTimeout(done, 10000)
+      })
+    })
```
Comment (dkarski) [inline packages/e2e/specs/overview/check-for-update.e2e.ts:11]: 🟢
Maybe move this piece of code as a helper with a meaningful name vs comment?

File: packages/e2e/.env.example [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[Required] The initial Mudita OS version on device
+TEST_INITIAL_VERSION=
+
+[Required] Path to Mudita Center binary
+TEST_BINARY_PATH=
+
+[Optional] WebdriverIO log level
+TEST_LOG_LEVEL=
```
Comment (dkarski) [inline packages/e2e/.env.example:8]:
```suggestion
# [Required] The initial Mudita OS version on device
TEST_INITIAL_VERSION=

# [Required] Path to Mudita Center binary
TEST_BINARY_PATH=

# [Optional] WebdriverIO log level
TEST_LOG_LEVEL=
```

Comment (dkarski) [reply packages/e2e/.env.example:8]: and what with `TEST_GITHUB_USER` & `TEST_GITHUB_PASSWORD` or `TEST_GITHUB_TOKEN`.

File: packages/e2e/specs/overview/display-initial-os-version.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  describe("Have initial OS version", () => {
+    before(async () => {
+      // Waiting for device connected through USB
+      await browser.executeAsync((done) => {
+        setTimeout(done, 10000)
+      })
+    })
+
+    it("should have valid OS version", async () => {
+      const VERSION_REGEXP = /[0-9].[0-9].[0-9]/
```
Comment (dkarski) [inline packages/e2e/specs/overview/display-initial-os-version.e2e.ts:13]: 🟡
100.0.0 - fails
1.0.02 - fails

What do you think to use `semver.valid` from `semver` lib? 🤔

File: packages/e2e/specs/overview/display-initial-os-version.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      const HARDCODED_VERSION = process.env.TEST_INITIAL_VERSION
```
Comment (dkarski) [inline packages/e2e/specs/overview/display-initial-os-version.e2e.ts:14]: I'm wondering if some fixture/script shouldn't set the initial version? 🤔

Comment (igorbezsmertnyi) [reply packages/e2e/specs/overview/display-initial-os-version.e2e.ts:14]: The initial version will be flashed to Pure with `pureflash` script. And this variable is needed just to, not change the code if for some reason we decided to bump the default version

File: packages/e2e/specs/overview/display-initial-os-version.e2e.ts (hunk 4 above)
Comment (igorbezsmertnyi) [reply packages/e2e/specs/overview/display-initial-os-version.e2e.ts]: https://github.com/mudita/mudita-center/pull/611/commits/b7e381a9b47bf9ffbf1f014e9fe4df4194fa8563

File: packages/e2e/.env.example (hunk 3 above)
Comment (igorbezsmertnyi) [reply packages/e2e/.env.example]: https://github.com/mudita/mudita-center/pull/611/commits/b7e381a9b47bf9ffbf1f014e9fe4df4194fa8563

File: packages/e2e/helpers/github.helper.ts (hunk 1 above)
Comment (igorbezsmertnyi) [reply packages/e2e/helpers/github.helper.ts:11]: https://github.com/mudita/mudita-center/pull/611/commits/b7e381a9b47bf9ffbf1f014e9fe4df4194fa8563

File: packages/app/src/tests/get-fake-adapters.ts [hunk 6]
```diff
@@ -1,42 +0,0 @@
-/**
```
Comment (dkarski) [inline packages/app/src/tests/get-fake-adapters.ts:1]: this is fails your check

**Review by dkarski — APPROVED** [review]

**Pull request merged**